explainer.md

```diff
@@ -147,17 +145,51 @@ function OnSessionStarted(session) {
+
+When it comes to ensuring canvas compatibility there's a two broad categories that apps will fall under.
+
+**VR Enhanced:** The app can take advantage of VR, but it's used as a progressive enhancement rather than a core part of the experience. Most users will probably not interact with the app's VR features, and as such asking them to make VR-centric decisions early in the app lifetime would be confusing and innapropriate. An example would be a news site with an embedded 360 photo gallery or video. (We expect the large majority of early WebVR content to fall into this category.)
+
+This style of application should call `VRDisplay.ensureContextCompatibility` with the WebGL context in question. This will set a compatibility bit on the context that allows it to be used. Contexts without the compatibility bit will fail when attempting to create a `VRLayer` with them. In the event that a context is not already compatible with the `VRDisplay` the [context will be lost and attempt to recreate itself](https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.13) using the compatible graphics adapter. It is the page's responsibility to handle WebGL context creation loss properly, recreating any necessary WebGL resources in response. If the context loss is not by the page handled the promise returned by `ensureContextCompatibility` will fail. The promise may also fail for a variety of other reasons, such as the context being actively used by a different, incompatible `VRDevice`.
+
+```js
+let gl = glCanvas.getContext("webgl");
+
+let vrLayer = null;
+
+function SetupCanvasLayer() {
+  // Make sure the canvas context we want to use is compatible with the device.
+  return vrDevice.ensureContextCompatibility(gl).then(() => {
+    // The content that will be shown on the device is defined by the session's
+    // baseLayer. In non-exclusive The baseLayer is not used for presentation, but
+    // the canvas dimensions are used to construct the projection matrices.
+    vrLayer = new VRCanvasLayer(vrSession, glCanvas);
+    vrSession.baseLayer = vrLayer;
+  });
 }
 ```
 
+**VR Centric:** The app's primary use case is VR, and as such it doesn't mind initializing resources in a VR-centric fashion, which may include asking users to select a headset as soon as the app starts. An example would be a game which is dependent on VR presentation and input. These types of applications can to avoid the need to call `ensureContextCompatibility` and the possible context loss that it may trigger by passing the `VRDevice` that the context will be used with as a context creation argument.
+
+```js
+let gl = glCanvas.getContext("webgl", { compatibleVrDevice: vrDevice });
+```
+
+Ensuring context compatibility with a `VRDisplay` through either method may have side effects on other graphics resources in the page, such as causing the entire user agent to switch from rendering using an integrated GPU to a discreet GPU.
```

Comment:
Is there a way to make this last sentence more visible? Since this could have a big impact, it's pretty important it not get lost in the explainer ^_^